manager/integration/tests/test_settings.py

```diff
@@ -1241,3 +1242,28 @@ def test_setting_update_with_invalid_value_via_configmap(core_api, request):  #
                        ""])
 
     cleanup_volume_by_name(client, vol_name)
+
+def test_setting_update_readonly():  # NOQA
```

Comment:
Can we use `client` fixture or `reset_settings` fixture, even if the test fails somehow, the settings will be set to the default values. No need to handle the setting to default value.

Reply:
We still allow modifying the read-only settings by `Longhorn APIs`.
I would change to modify settings by the manifest.
or wait for changing codes if we don't allow modifying the read-only settings by `Longhorn APIs`. cc @innobead

Reply:
Modified to use the patching custom object API.